### PR TITLE
Add `plot()` functionality to audio metrics

### LINKED ISSUE #1406: Add `.plot()` to all modular metrics
## 🚀 Feature

We are in the process of adding native plotting to torchmetrics such that our users can easily create simple plots of metrics, which is especially useful for metrics such as confusion matrix and pr-curves but even single-scalar metrics such as accuracy it can be beneficial to quickly create a plot of computed values.

The first proof of concept PR has just been merged: https://github.com/Lightning-AI/metrics/pull/1328

We are inviting anyone interested in contributing to torchmetrics to help add this feature to all remaining metrics. The tasks are fairly simple and an example of how to implement the `plot` metric can be seen here: https://github.com/Lightning-AI/metrics/blob/d8caeeb28efc0afb86ae68cf66b06412749a4e46/src/torchmetrics/classification/accuracy.py#L110-L156

We would like the work divided into PRs tackling each domain, however if any are interested in only contributing to a few metrics within a particular domain that is also more than welcome.

- [ ] Aggregation ->#1485 @SkafteNicki 
- [ ] Audio ->#1434 @shhs29
- [ ] Classification -> @srishti-git1110
- [ ] Image ->#1480 @venomouscyanide
- [ ] Detection
- [ ] Regression -> @srishti-git1110
- [ ] Retrieval -> @ufukhurriyetoglu
- [ ] Text -> @ufukhurriyetoglu

### PR DESCRIPTION
**Pull request by shhs29**
## What does this PR do?

Fixes #1406

Currently, plot functionality is added for pesq metric. The test for plotting passes locally. However, the workflows on CI are failing.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki @justusschock mind have look :otter:

**Comment by SkafteNicki** [comment]
Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :)
Thanks for your work.

**Comment by shhs29** [comment]
> Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.

@SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.

**Comment by SkafteNicki** [comment]
> > Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.
> 
> @SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.

No problem. Just tag me when you are done with the rest and I can help cleanup and fix anything that may be missing/fails :]

**Comment by shhs29** [comment]
> > > Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.
> > 
> > 
> > @SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.
> 
> No problem. Just tag me when you are done with the rest and I can help cleanup and fix anything that may be missing/fails :]

@SkafteNicki I have added plot functionality for all the audio metrics. However, I am unsure about the lower and upper bounds of these metrics. Also, a couple of Github actions are failing, which I suspect is due to some incorrect/insufficient imports. I would appreciate your feedback on the same.

File: src/torchmetrics/audio/stoi.py [hunk 1]
```diff
@@ -11,16 +11,20 @@
...
 from torchmetrics.functional.audio.stoi import short_time_objective_intelligibility
 from torchmetrics.metric import Metric
-from torchmetrics.utilities.imports import _PYSTOI_AVAILABLE
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _PYSTOI_AVAILABLE
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE, plot_single_or_multi_val
 
 __doctest_requires__ = {"ShortTimeObjectiveIntelligibility": ["pystoi"]}
 
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["SignalNoiseRatio.plot", "ScaleInvariantSignalNoiseRatio.plot"]
```
Comment (SkafteNicki) [inline src/torchmetrics/audio/stoi.py:26]:
```suggestion
    __doctest_skip__ = ["ShortTimeObjectiveIntelligibility.plot"]
```

**Comment by SkafteNicki** [comment]
@shhs29 thanks, I will take a look at it soon. We are experience problems with our CI at the moment, which should hopefully be solved soon also.

File: src/torchmetrics/audio/pesq.py [hunk 2]
```diff
@@ -119,3 +124,50 @@ def update(self, preds: Tensor, target: Tensor) -> None:
...
+        """Plot a single or multiple values from the metric.
+
+        Args:
+            val: Either a single result from calling `metric.forward` or `metric.compute` or a list of these results.
+                If no value is provided, will automatically call `metric.compute` and plot that result.
+            ax: An matplotlib axis object. If provided will add plot to that axis
+
+        Returns:
+            fig: Figure object
+            ax: Axes object
```
Comment (Borda) [inline src/torchmetrics/audio/pesq.py:140]:
```suggestion
        Returns:
            Figure and Axes object
```

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**